### PR TITLE
feat(iroh): add blobs.create_collection api endpoint

### LINKED ISSUE #1950: Feature Request: New RPC Method: `iroh.blobs.add_collection(hashes: BTreeMap<String, Hash>)`
Working over RPC, there's no way to replicate https://github.com/n0-computer/iroh/blob/eab55bf97acff7df3163240c7dc438520e4874f0/iroh/src/node.rs#L1248-L1279 , because we don't have access to `self.inner.db`

I think it would be nice to have a new method on the `blobs` RPC command:

```rust
iroh.blobs.add_collection(meta: BTreeMap<String,Hash>, tag: ) -> Result<Hash>
```

This would error if any of the hashes in the collection isn't already present in the store (except fro the meta & collection root blob), and return the root hash of the collection

### PR DESCRIPTION
**Pull request by dignifiedquire**
Closes #1950

### CODE REVIEW

File: iroh/src/rpc_protocol.rs [hunk 1]
```diff
@@ -274,6 +274,28 @@ impl RpcMsg<ProviderService> for DeleteTagRequest {
...
+    pub tag: SetTagOption,
+}
+
+/// A response to a create collection request
+#[derive(Debug, Serialize, Deserialize)]
+pub struct CreateCollectionResponse {
+    /// The resulting hash.
+    pub hash: Hash,
+    /// The resulting tag.
+    pub tag: Tag,
```
Comment (rklaehn) [inline iroh/src/rpc_protocol.rs:292]: 👍

**Review by rklaehn — APPROVED** [review]
This works as intended, but I am not sure it is a great API.

The only way to create a collection is to make multiple individual blobs that all have a tag so they don't get immediately deleted, then create a collection out of them. But then you got all these tags that you probably don't want anymore.

Ideally we would have a way to extend the concept of temp tags over the iroh/iroh-client boundary. But maybe that can be done in a subsequent PR.

**Comment by dignifiedquire** [comment]
> Ideally we would have a way to extend the concept of temp tags over the iroh/iroh-client boundary. But maybe that can be done in a subsequent PR.

Yes I agree, but I wanted to have this functionality sooner rather than later, as designing a solution for `TempTag`s over RPC is non trivial.

**Comment by rklaehn** [comment]
> > Ideally we would have a way to extend the concept of temp tags over the iroh/iroh-client boundary. But maybe that can be done in a subsequent PR.
> 
> Yes I agree, but I wanted to have this functionality sooner rather than later, as designing a solution for `TempTag`s over RPC is non trivial.

Understood. People want to use this. But let's revisit it at some point.

**Comment by Frando** [comment]
> Ideally we would have a way to extend the concept of temp tags over the iroh/iroh-client boundary. But maybe that can be done in a subsequent PR.

Yes, this would be great to have I think. Currently the docs RPC API copies things from the blobs api only because otherwise we could not add entries to docs without each having a tag. We could remove this potentially if temp tags could cross the RPC boundary.

**Pull request merged**